### DISCUSSION TITLE
Better rendering for timeline elements

### DISCUSSION THREAD

**Discussion by pzljanb** [Puzzlepart/prosjektportalen365#586] [Ideas]
**Is your feature request related to a problem? Please describe.**
Elements are overlapping in timeline view. And of that reason hard to separate/ read.

**Describe the solution you'd like**
1. There should be a required gap between element1 stop date and element2 start date before putting 2 elements on the same line within the timeline view.
A gap of 7 days could be practical to make sure the description is not covering the second element.
2. Text colour of description (white as of today) is not practical on white background. The colour should be change to give better contrast.

**Additional context**

![image](https://user-images.githubusercontent.com/58074707/137923863-91a0ffcd-077c-4ac9-afe6-70781c818276.png)

![image](https://user-images.githubusercontent.com/58074707/137923847-0a289595-4f25-4339-8a23-4193ebd01e02.png)

![image](https://user-images.githubusercontent.com/58074707/137923815-6dffbce6-097d-4500-9cec-95b38e13d78c.png)

![image](https://user-images.githubusercontent.com/58074707/137923763-6835b6ca-0bbd-43a9-9707-3cdafb7c2cad.png)

**Comment by okms** [comment]
As I understand it the problem arises with consecutive events, not overlapping events? Can we force alternating colors for events on the same plane in the timeline? If not, how about a clearer separator? This will of course not solve the issue of descriptions overlapping, but it will make the distinction between two different events running back-to-back easier.

**Comment by pzljanb** [comment]
> As I understand it the problem arises with consecutive events, not overlapping events? Can we force alternating colors for events on the same plane in the timeline? If not, how about a clearer separator? This will of course not solve the issue of descriptions overlapping, but it will make the distinction between two different events running back-to-back easier.

Consecutive planning of elements gives overlapping in graphical UI

**Comment by okms** [comment]
Ok, it's a bit hard to tell, but i can see there is a 5px overlap/overflow. "Fase 2" should have started at the very least 5px more to the right so its on the right side of the yellow marker

![image](https://user-images.githubusercontent.com/170147/137932400-49c0a23b-a50f-4991-a7f9-9688cca67661.png)

**Reply by olemp** [reply]
Converted to discussion @pzljanb @tarjeieo @okms. 

This is not something we can easily adjust when using a 3rd party library, but let's keep a discussion open.
